### ISSUE TITLE
github issue overhaul

### ISSUE THREAD

**Issue by BrittanyBunk** [internetarchive/openlibrary#6076]
**Warnings:**
- I'm putting this issue here for documentation purposes, as it's a common issue among multiple members. It's not a complaint or blaming of anyone, but a hope for better - which is why this is a community discussion.
- this issue is a much lower priority than other issues, so this should get less focus on than others.
- I might've already posted this before, but this one's slightly different.

**Problem:**
- github is insufficient for addressing needs and is problematic in its organization (one example is that there's no 'neutral' action on an issue - either open or closed. This causes issues where problems that're good but are not a priority are closed rather than be 'inactive'. Just one problem of the many)
- volunteers don't have access to certain capabilities needed to work on issues nor are allowed to work on issues, which creates a bottleneck for getting work done. There's a 'lead' label, but the workflow process is disruptive to the process.
- it was brought up during the community meeting that staff members don't get notifications from github on updates or new issues - so they can't work on them properly.
- the github repository gets overloaded and many issues fail to be worked on or get worked on when they shouldn't have to.

**Possible solutions:**
- find a better platform than github - let's do a community discussion on this
- make github more streamlined to be worked on
- find an email system that can forward messages via email properly
- decentralize volunteering to allow for volunteers to work on projects, so staff doesn't have to as much
- decentralize communication to not need to go to staff as much to get everything done
- create a new repository for the OL that can separate what the staff works on from what volunteers work on - maybe have a volunteer repository? Where volunteers work with other volunteers on issues that don't require staff, so that it keeps their workload small by avoiding the need to look at notifications and the need for volunteers to wait on staff to move forward.
- Maybe a different repository than a volunteer one? Let's discuss this

**Constraints:**
- there're plenty of volunteer outlets already created for volunteers to discuss issues at
- volunteer discussion locations have been terminated due to the risks of abuse
- volunteers might end up wasting their time if they're working on issues that're never going to be picked up by the Open Library
- having more repositories will be harder to track and might be more work to notify everyone.

I'm opening the brainstorm discussion, so we can come up with a better volunteer/github issue/solving distribution to work out something better.

**Comment by bicolino34** [comment]
@BrittanyBunk as far as I know, there is no better platform for programming than GitHub
> find an email system that can forward messages via email properly

What do you mean by properly? It works perfect for me. It sends me message about new comments on issues I'm participating in or about mentions in issues. You can easily change settings to receive e-mail about every single new comment in this repo.

**Comment by BrittanyBunk** [comment]
@bicolino34 I think there's a misunderstanding. I agree - github is the best for sharing computer programs, but to me, sharing and solving problems is where it's not seamless like sharing code is on here. Even though github does fall short for even sharing computer code (sometimes that's why jupyter notebooks and other platforms are experimented with - to document code in a journal-like format - which has its own pros/cons), that's not what I'm after. I'm after a better platform for the process of problems -> solutions, along with brainstorming. A place where people can see each process, like a mind map and see what went where and what was at the end of each idea path - a dead end, or a solution - and what's the solution? Who did what at which part in the path? Having it as comments, like how github and other social media platforms isn't visual and connective like this, and that's why social media has so many problems (imo - in my opinion). Github tries to act like social media when it shouldn't. I want to look/seek out a decision-making, brainstorming platform that has the features I described - a visual flowchart setup, not a social media commenting one.

@bicolino34 this is what I heard from the staff. I heard they get too many notifications, but I think that's why it's even more important to do a visual setup, because people can jump in at the fragment ends of paths to where they don't really need notifications. Also since others can work out each path, it'll lead to even fewer notifications.

But from my experience, I only get notifications when someone tags me. That means if someone replies back to me without tagging me, I don't see it. This is not only an issue on github, but other places too.

Maybe I'm making a big deal of nothing, but personally I do want to know if better does exist, so that I can be ready with an answer when asked for better, as I feel it's going to come up in the future - even if it doesn't for the OL, it could help for my work on it.

**Comment by mekarpeles** [comment]
1. Emails are forwarded, the volume is too great and so notifications are muted. It would be great if there were more granular notification subscription options. If someone knows how to do this well, please do reply!
2. I don't want to entertain the idea of switching off of github, too many of our workflows are tied to it. There's lots we can do to augment our process.
3. As posted in #6084, we have a new process we're trying in order to do a better job addressing this problem. Each week we'll run a query which shows us the issues which have been commented most recently. Each team lead will take responsibility for making sure those issues are read, evaluated/triaged, and (if if appropriate + there's been an offer) assign someone from the community.

Archiving this issue and thankful as it exists -- useful context!

**Comment by BrittanyBunk** [comment]
@mekarpele thanks for appreciating the context and documentation purposes!

1) It's actually really easy! I get issues with email notifications and then I handle them, so currently I have none and I address everyone just fine.

Solutions I do:
1) filter notifications - you create 'rules' that filter out notifications by keyword. So if you get notifications that someone read a github post and that's not important, you create a rule saying to filter out notifications that have the words 'someone read' in it. Warning: it might delete legitimate emails, but that would be 5% of them, so if it happens, other people will let you know.
2) look at the settings on https://github.com/settings/emails - maybe some settings can be turned off (you might've done this already, but still)
3) go back to the muted notifications and actually read them - to find important ones there!
4) do a 'sanity check' (your words, not mine) - ask others what you should be looking for

-----
3) glad you have some ideas to help it along. I think it's an awesome idea you have and am glad you're going to try it out. It's not a complete solution (i.e. streamlining), as I said before, but it will really help something somewhere and sometimes that's all we need to get by in life!
